common/client-core/src/client/statistics_control.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+//! # Statistics collection and reporting.
+//!
+//! Modular metrics collection and reporting system. submodules can be added to collect different types of metrics.
+//! On creation the Statistics controller will start a task that will listen for incoming stats events and
+//! multiplex them out to the appropriate metrics module based on type.
+//!
+//! Adding A new module you need to write a new module that implements the `StatsObj` trait and add it to
+//! the `stats` hashmap in the `StatisticsControl` struct during it's initialization in the `new` function in
+//! this file.
+
+#![warn(clippy::expect_used)]
+#![warn(clippy::unwrap_used)]
+#![warn(clippy::todo)]
+#![warn(clippy::dbg_macro)]
+
+use std::time::Duration;
+
+use nym_sphinx::addressing::Recipient;
+use nym_statistics_common::clients::{
+    ClientStatsController, ClientStatsReceiver, ClientStatsSender,
+};
+use nym_task::connections::TransmissionLane;
+
+use crate::{
+    client::inbound_messages::{InputMessage, InputMessageSender},
+    spawn_future,
+};
+
+/// Time interval between reporting statistics
+const STATS_REPORT_INTERVAL_SECS: u64 = 300;
```

Comment:
rather than having opaque local constants, why not put them explicitly in client's config (in one of `debug` sections) so that they could be adjusted more easily should we need it?

Reply:
Got it, I will integrate that along with the address distribution in .env file in the next PR